compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -77,6 +82,36 @@ generateSrcFromLf env = noLoc mod
         let occName = mkOccName varName $ T.unpack $ sanitize dataTypeCon0
         pure $ mkDataDecl env thisModule occName dataParams (convDataCons dataTypeCon0 dataCons)
 
+    valueDecls = do
+        dval@LF.DefValue {..} <- NM.toList $ LF.moduleValues $ envMod env
+        guard $ shouldExposeDefValue dval
+        let (lfName, lfType) = dvalBinder
+            ltype = noLoc $ convType env lfType :: LHsType GhcPs
+            lname = mkRdrName (LF.unExprValName lfName) :: Located RdrName
+            sig = TypeSig noExt [lname] (HsWC noExt $ HsIB noExt ltype)
+            lsigD = noLoc $ SigD noExt sig :: LHsDecl GhcPs
+            lexpr = noLoc $ HsPar noExt $ noLoc $ HsVar noExt lname :: LHsExpr GhcPs
+            lgrhs = noLoc $ GRHS noExt [] lexpr :: LGRHS GhcPs (LHsExpr GhcPs)
+            grhss = GRHSs noExt [lgrhs] (noLoc $ EmptyLocalBinds noExt)
+            matchContext = FunRhs lname Prefix NoSrcStrict
+            lmatch = noLoc $ Match noExt matchContext [] Nothing grhss
+            lalts = noLoc [lmatch]
+            bind = FunBind noExt lname (MG noExt lalts Generated) WpHole []
+            lvalD = noLoc $ ValD noExt bind :: LHsDecl GhcPs
+        [ lsigD, lvalD ]
+
+    shouldExposeDefDataType :: LF.DefDataType -> Bool
+    shouldExposeDefDataType typeDef
+        = not (defDataTypeIsOldTypeClass typeDef)
+
+    shouldExposeDefValue :: LF.DefValue -> Bool
+    shouldExposeDefValue LF.DefValue{..}
+        | (lfName, lfType) <- dvalBinder
+        = not (LF.getIsTest dvalIsTest)
```

Comment:
Is there actually a problem with exposing scenarios? I could see it being useful, e.g., if you have some function to initialize the state in scenarios you might want to use it across upgrades.

Reply:
I'll expose them in a separate PR :)